tests/cpp/test_loop_domain_scheduling.cpp

```diff
@@ -317,4 +317,67 @@ TEST_F(LoopDomainSchedulingTest, ManyReshape) {
   }
 }
 
+// Testing scheduleLoopDomainsBy with a trivial fusion
+TEST_F(LoopDomainSchedulingTest, ScheduleLoopDomainsBy) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  std::vector<int64_t> shape({100});
+
+  // concrete shapes to avoid dynamic Fusion
+  auto tv0 = makeConcreteTensor(shape);
+  fusion.addInput(tv0);
+
+  auto tv1 = unaryOp(UnaryOpType::Sin, tv0);
+  auto tv2 = unaryOp(UnaryOpType::Cos, tv1);
+  auto tv3 =
+      slice(tv2, {{IrBuilder::create<Val>(1L), IrBuilder::create<Val>(99)}});
+  auto tv4 = unaryOp(UnaryOpType::Exp, tv3);
+
+  fusion.addOutput(tv4);
+
+  auto resize = tv3->getLogicalDomain().at(0)->definition()->as<Resize>();
+
+  scheduler_tools::scheduleLoopDomainsBy({tv1, tv2, tv4}, resize);
+
+  // tv1 and tv2 should have the same loop domain as tv3's loop domain
+  IdModel id_model(&fusion, /*build_graphs=*/false);
+  const auto& exact_graph = id_model.buildExactGraph();
+  EXPECT_EQ(
+      exact_graph.toGroups(tv1->getLoopDomain()),
+      exact_graph.toGroups(tv3->getLoopDomain()));
+  EXPECT_EQ(
+      exact_graph.toGroups(tv2->getLoopDomain()),
+      exact_graph.toGroups(tv3->getLoopDomain()));
```

Comment:
Should we also check that `tv1->axis(0)->definition()->isA<Resize>()` and similar for `tv2`?

Reply:
I don't think it's necessary because if the loop IDs are exact mapped, `tv1` should have the same defining expr as the loop domain of `tv3`.